deeppavlov/skills/dsl_skill/dsl_skill.py

```diff
@@ -0,0 +1,188 @@
+
+    Example:
+
+    .. code:: python
+
+            class ExampleSkill(metaclass=ZDialog):
+                @ZDialog.handler(commands=["hello", "hey"], state="greeting")
+                def __greeting(message: str):
+                    ...
+    """
+    skill_collection: Dict[str, 'DSLMeta'] = {}
+
+    def __init__(cls, name: str,
+                 bases,
+                 namespace,
+                 **kwargs):
+        super(DSLMeta, cls).__init__(name, bases, namespace, **kwargs)
+        cls.name = name
+
+        # Attribute cls.state_to_handler is dict with states as keys and lists of Handler objects as values
+        cls.state_to_handler = defaultdict(list)
+        # Handlers that can be activated from any state
+        cls.universal_handlers = []
+
+        handlers = [attribute for attribute in namespace.values() if isinstance(attribute, Handler)]
+
+        for handler in handlers:
+            if handler.state is None:
+                cls.universal_handlers.append(handler)
+            else:
+                cls.state_to_handler[handler.state].append(handler)
+
+        cls.handle = partial(DSLMeta.__handle, cls)
+        cls.__call__ = partial(DSLMeta.__handle_batch, cls)
+        cls.__init__ = partial(DSLMeta.__init__class, cls)
+        register()(cls)
+        DSLMeta.__add_to_collection(cls)
+
+    def __init__class(cls, on_invalid_command: str = "Простите, я вас не понял",
+                      null_confidence: float = 0,
+                      *args, **kwargs):
+        """
+        Initialize Skill class
+        Args:
+            on_invalid_command:  message to be sent on message with no associated handler
+        """
+        # message to be sent on message with no associated handler
+        cls.on_invalid_command = on_invalid_command
+        cls.null_confidence = null_confidence
+
+    def __handle_batch(cls: 'DSLMeta',
+                       utterances_batch: List,
+                       history_batch: List = None,
+                       states_batch: List = None) -> Tuple[List, ...]:
+        """Returns skill inference result.
+        Returns batches of skill inference results, estimated confidence
+        levels and up to date states corresponding to incoming utterance
+        batch.
+        Args:
+            utterances_batch: A batch of utterances of str type.
+            history_batch: A batch of list typed histories for each utterance.
+            states_batch:  A batch of arbitrary typed states for
+                each utterance.
+        Returns:
+            response_batch: A batch of arbitrary typed skill inference results.
+            confidence_batch: A batch of float typed confidence levels for each of
+                skill inference result.
+            output_states_batch:  A batch of arbitrary typed states for
+                each utterance.
+
+        """
+        history_batch = history_batch or []
```

Comment:
I like the original better

Reply:
There are lots of examples how to handle the case of default `None` argument which should be converted to empty `List` in official and non-official python documentation. For example, here:
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html
here:
https://docs.python-guide.org/writing/gotchas/
and, in official docs, here:
https://docs.python.org/3/reference/compound_stmts.html#function-definitions
It's done like this because explicit is better than implicit (zen of python) and because it's less error prone, since this code will work not only when `history_batch` is `None`.
If you want a one-liner, use `[] if history_batch is None else history_batch`